Adding Failure Listener documentation for lottie-android

## lottie-android App crash on invalid json url [#2271](https://github.com/airbnb/lottie-android/issues/2271)

I came across a crash issue on lottie-android, user was using invalid json url in `LottieAnimationView` using `lottie_url` property. That is causing an app crash with exception:


```java
E  FATAL EXCEPTION: main
Process: com.example.lottieandroidissue2271, PID: 25682
java.lang.IllegalStateException: Unable to parse composition
	at com.airbnb.lottie.LottieAnimationView.lambda$static$0(LottieAnimationView.java:74)
	at com.airbnb.lottie.LottieAnimationView$$ExternalSyntheticLambda2.onResult(Unknown Source:2)
	at com.airbnb.lottie.LottieAnimationView$1.onResult(LottieAnimationView.java:86)
	at com.airbnb.lottie.LottieAnimationView$1.onResult(LottieAnimationView.java:79)
	at com.airbnb.lottie.LottieTask.notifyFailureListeners(LottieTask.java:162)
	at com.airbnb.lottie.LottieTask.lambda$notifyListeners$0$com-airbnb-lottie-LottieTask(LottieTask.java:138)
	at com.airbnb.lottie.LottieTask$$ExternalSyntheticLambda0.run(Unknown Source:2)
	at android.os.Handler.handleCallback(Handler.java:883)
	at android.os.Handler.dispatchMessage(Handler.java:100)
	at android.os.Looper.loop(Looper.java:237)
	at android.app.ActivityThread.main(ActivityThread.java:8167)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:496)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1100)
Caused by: com.airbnb.lottie.parser.moshi.JsonEncodingException: Use JsonReader.setLenient(true) to accept malformed JSON at path $
	at com.airbnb.lottie.parser.moshi.JsonReader.syntaxError(JsonReader.java:251)
	at com.airbnb.lottie.parser.moshi.JsonUtf8Reader.checkLenient(JsonUtf8Reader.java:951)
	at com.airbnb.lottie.parser.moshi.JsonUtf8Reader.doPeek(JsonUtf8Reader.java:355)
	at com.airbnb.lottie.parser.moshi.JsonUtf8Reader.beginObject(JsonUtf8Reader.java:144)
	at com.airbnb.lottie.parser.LottieCompositionMoshiParser.parse(LottieCompositionMoshiParser.java:56)
	at com.airbnb.lottie.LottieCompositionFactory.fromJsonReaderSyncInternal(LottieCompositionFactory.java:393)
	at com.airbnb.lottie.LottieCompositionFactory.fromJsonReaderSync(LottieCompositionFactory.java:386)
	at com.airbnb.lottie.LottieCompositionFactory.fromJsonInputStreamSync(LottieCompositionFactory.java:329)
	at com.airbnb.lottie.LottieCompositionFactory.fromJsonInputStreamSync(LottieCompositionFactory.java:322)
	at com.airbnb.lottie.network.NetworkFetcher.fromJsonStream(NetworkFetcher.java:150)
	at com.airbnb.lottie.network.NetworkFetcher.fromInputStream(NetworkFetcher.java:123)
	at com.airbnb.lottie.network.NetworkFetcher.fetchFromNetwork(NetworkFetcher.java:84)
	at com.airbnb.lottie.network.NetworkFetcher.fetchSync(NetworkFetcher.java:45)
	at com.airbnb.lottie.LottieCompositionFactory.lambda$fromUrl$0(LottieCompositionFactory.java:128)
	at com.airbnb.lottie.LottieCompositionFactory$$ExternalSyntheticLambda3.call(Unknown Source:6)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:919)
```


Adding a failure listener will not crash the whole application, instead it will call the failure listener's `onResult` function. Although exception will come in onResult but then its user's preference to retry or `setFallbackRes`, once an exception occurs.


```java
onResult: 
com.airbnb.lottie.parser.moshi.JsonEncodingException: Use JsonReader.setLenient(true) to accept malformed JSON at path $
	at com.airbnb.lottie.parser.moshi.JsonReader.syntaxError(JsonReader.java:251)
	at com.airbnb.lottie.parser.moshi.JsonUtf8Reader.checkLenient(JsonUtf8Reader.java:951)
	at com.airbnb.lottie.parser.moshi.JsonUtf8Reader.doPeek(JsonUtf8Reader.java:355)
	at com.airbnb.lottie.parser.moshi.JsonUtf8Reader.beginObject(JsonUtf8Reader.java:144)
	at com.airbnb.lottie.parser.LottieCompositionMoshiParser.parse(LottieCompositionMoshiParser.java:56)
	at com.airbnb.lottie.LottieCompositionFactory.fromJsonReaderSyncInternal(LottieCompositionFactory.java:393)
	at com.airbnb.lottie.LottieCompositionFactory.fromJsonReaderSync(LottieCompositionFactory.java:386)
	at com.airbnb.lottie.LottieCompositionFactory.fromJsonInputStreamSync(LottieCompositionFactory.java:329)
	at com.airbnb.lottie.LottieCompositionFactory.fromJsonInputStreamSync(LottieCompositionFactory.java:322)
	at com.airbnb.lottie.network.NetworkFetcher.fromJsonStream(NetworkFetcher.java:150)
	at com.airbnb.lottie.network.NetworkFetcher.fromInputStream(NetworkFetcher.java:123)
	at com.airbnb.lottie.network.NetworkFetcher.fetchFromNetwork(NetworkFetcher.java:84)
	at com.airbnb.lottie.network.NetworkFetcher.fetchSync(NetworkFetcher.java:45)
	at com.airbnb.lottie.LottieCompositionFactory.lambda$fromUrl$0(LottieCompositionFactory.java:128)
	at com.airbnb.lottie.LottieCompositionFactory$$ExternalSyntheticLambda3.call(Unknown Source:6)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:923)
```

This is the default behaviour of lottie-android library, so I think it should be in the documentation, I have forked this repo and added a documentation for Failure Listener.

You can check my repo: [lottie-android-issue-2271-doc-update](https://github.com/MSVhora/lottie/tree/lottie-android-issue-2271-doc-update)

--- a/android.md
+++ b/android.md
@@ -107,6 +107,14 @@ In the update listener callback:
 
 `animation.getAnimatedFraction()` will return the progress of the animation taking into account the set min/max frame [minFrame,maxFrame].
 
+### Failure Listener
+
+Set a default failure listener that will be called if any of the setAnimation APIs fail for any reason. This can be used to replace the default behavior. The default behavior will log any network errors and rethrow all other exceptions.
+
+If you are loading an animation from the network, errors may occur if your user has no internet. You can use this listener to retry the download or you can have it default to an error drawable with `setFallbackResource(int)`. Unless you are using `setAnimationFromUrl(String)`, errors are unexpected.
+
+Set the listener to `null` to revert to the default behavior.
+
 ### Custom animators
 
 Although `playAnimation()` is sufficient for the vast majority of use cases, you can call `setProgress(...)` in the update callback for your own animator. This can be useful to tie an animation to something like a gesture, download progress, or scroll position.

